common/lib/CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ add_library(common_crt OBJECT
    divmod.cc
    mul.cc
    shift.cc
+
+   new.cc
```

Comment:
As is, this doesn't really fit in the C runtime library, since it's not a part of the C runtime; it's a part of the C++ runtime.

For LLVM, implementations of most of these functions appear to be in their libcxx; these should probably be in a libc++ that forms the C++ runtime and standard library.

Generally, it's expected to have a libc++ depend on a libc, but not the other way around. Accordingly, we can move the actual heap implementation into libc, and only reference it from libc++. We can make the compiler driver  automatically import our libc++ if called via clang++, as is the usual practice. This keeps the C linker namespace clean of mangled C++ symbols like _ZNewWhatever.

This doesn't matter all that much, but it's generally important to carefully allocate a project's "surprise points." If there's a really specific reason to implement the heap in new.cc, let me know; we can then document that and have an explanation if folks come asking about it.

Reply:
Makes sense; we do probably have some pre-work to do to establish a libc++. At the very least, I'll want to survey other compiler targets to get a feel for naming conventions; I've seen libcpp, libcppabi, libc++, libstdc++.

For now, if you would, move this stuff to libc. It definitely doesn't belong in libcrt (which is exclusively for compiler-internal libcalls), and we'll still probably want the malloc stuff to stay in libc when we get around to building a real libc++.